src/libs/ajax/methods/providers/PermissionsProvider.ts

```diff
@@ -0,0 +1,70 @@
+import { AbortOption } from '@terra-ui-packages/data-client-core';
+import { Methods } from 'src/libs/ajax/methods/Methods';
+import { MethodConfigACL } from 'src/libs/ajax/methods/methods-models';
+
+export interface PermissionsProvider {
+  getPermissions: (
+    namespace: string,
+    name?: string | undefined,
+    snapshotId?: number | undefined,
+    options?: AbortOption
+  ) => Promise<MethodConfigACL>;
+  updatePermissions: (
+    namespace,
+    updatedPermissions: MethodConfigACL,
+    name?: string | undefined,
+    snapshotId?: number | undefined,
+    options?: AbortOption
+  ) => Promise<MethodConfigACL>;
+}
+
+/**
+ * Permissions provider to fetch and update the namespace permissions in Broad Methods Repository
+ */
+export const namespacePermissionsProvider: PermissionsProvider = {
+  getPermissions: async (
+    namespace: string,
+    _name?: string | undefined,
+    _snapshotId?: number | undefined,
```

Comment:
Yes for namespace permissions editing, both `name` and `snapshotId` are not required. But since the PermissionsProvider is the same for both namespace and snapshot I need to have it here in the function parameters. As a result I have marked both `name` and `snapshotId` with `_` which indicated that they are "purposefully not being used"